js/main.js

```diff
@@ -169,21 +185,26 @@ class App extends React.Component {
     if (this._socket) {
       return;
     }
-
     var url = window.location;
     var socket = new WebSocket('ws://' + url.host + this.correctPathname() + 'socket');
 
     socket.onmessage = this._handleMessage;
 
     socket.onopen = () => {
-      this.setState({connected: true});
-    }
-
-    socket.onerror = socket.onclose = () => {
-      this.setState({connected: false}, () => {
+      this.setState({connected: true}, () => {console.log('callback');});
```

Comment:
Nit: Remove callbacks once migrating to `componentDidUpdate` for the functionality you wanted, then you can leave this socket section how it was.